vasprintf_compat.h

```diff
@@ -51,7 +51,7 @@ static int vasprintf(char **buf, const char *fmt, va_list ap)
 		return -1;
 	}
 
-	if ((chars = vsprintf(b, fmt, ap)) < 0)
+	if ((chars = vsnprintf(b, ((size_t)chars + 1), fmt, ap)) < 0)
```

Comment:
The code just above this already uses vsprintf to calculate the right length and allocate the memory needed.  Passing a length just adds unnecessary checking.